### ISSUE TITLE
lines summary counts a statementMap entry that has no counter, istanbul does not

### ISSUE THREAD

**Issue by BartWaardenburg** [fallow-rs/oxc-coverage-instrument#173]
The `lines` metric diverges from `istanbul-lib-coverage` when a `statementMap` entry has no matching `s` counter. We report the line as uncovered, istanbul does not report the line at all.

### Reproduction

`cov.json`, where `statementMap` holds ids `0` and `1` but `s` holds only id `0`:

```json
{"a.js":{"path":"a.js",
"statementMap":{"0":{"start":{"line":1,"column":0},"end":{"line":1,"column":5}},
                "1":{"start":{"line":7,"column":0},"end":{"line":7,"column":5}}},
"fnMap":{},"branchMap":{},
"s":{"0":3},
"f":{},"b":{}}}
```

istanbul-lib-coverage:

```js
const { createCoverageMap } = require("istanbul-lib-coverage");
createCoverageMap(require("./cov.json")).fileCoverageFor("a.js").toSummary().lines;
// { total: 1, covered: 1, skipped: 0, pct: 100 }
```

Ours:

```
oxc-coverage-instrument report --format json-summary cov.json
// lines: { "total": 2, "covered": 1, "skipped": 0, "pct": 50.0 }
```

100% against 50% on the same input.

### Cause

`lines_metric` in `crates/oxc_coverage_report/src/summary.rs` iterates `statement_map` and resolves the counter with `file.s.get(id).unwrap_or(0)`, so an entry with no counter is added to `all_lines` as uncovered.

`getLineCoverage` in `istanbul-lib-coverage/lib/file-coverage.js` iterates the `s` counter map instead, and skips any id that is missing from `statementMap`:

```js
Object.entries(statements).forEach(([st, count]) => {
    if (!statementMap[st]) {
        return;
    }
    ...
});
```

An entry present in `statementMap` but absent from `s` is therefore never visited and contributes no line.

### Why this is reachable

This is not a theoretical shape. #114 covers a case where a `statementMap` entry survives while its `++cov.s[N]` is dropped from the emitted code, which produces exactly this input.

### Open question

Counting the line as uncovered is arguably the safer reading, so this may be a divergence worth keeping rather than a defect to fix. Either way it should be a decision: match istanbul, or document the divergence.

`missing_statement_and_function_counters_are_uncovered` already feeds this input but asserts only `statements` and `functions`, so whichever behaviour is chosen needs a `lines` assertion to pin it.

### Version

0.10.3
